### DISCUSSION TITLE
How To Replicate SQL COALESCE Across Multiple Data Sources

### DISCUSSION THREAD

**Discussion by ollie-acuto** [prisma/prisma#20580] [Q&A]
### Question

I am looking to select data in a prisma query utilising relations across 3 models in my schema file. I need to populate 3 custom fields (as alias) with data from one model if not null, and from another model if null. How can I do this with prisma?

This is a raw SQL sample query that I need to replicate:

```sql
SELECT 
    properties.id AS id,
    company_properties.external_id AS external_id,
    COALESCE(properties_edited.name, company_properties.name) AS name,
    COALESCE(properties_edited.address, properties.address) AS address,
    COALESCE(properties_edited.state, properties.state) AS state,
    company_properties.source AS source
FROM 
    company_properties 
    LEFT JOIN properties 
        ON company_properties.property_id = properties.id
    LEFT JOIN properties_edited 
        ON properties.id = properties_edited.property_id
WHERE 
    (company_properties.company_id = ? 
        AND company_properties.status = "active" 
        AND (company_properties.name LIKE ? OR company_properties.source LIKE ?)
    ) 
ORDER BY
    ?? DESC
LIMIT ? OFFSET ?;
```

I have tried using `$queryRaw` and `$queryRawUnsafe`, but need to dynamically insert a column name into the `ORDER BY` clause, which is not supported. 

I have also looked into adding custom fields to query result through [client extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result), but it looks like this can't be done across multiple models or in conjunction with existing queries (e.g. findMany). Am I missing anything here?

Would appreciate any help here!

Environment & setup
macOS
MySQL
Node 16.20.0

Prisma Version
"@prisma/client": "^4.16.1"

**Comment by ludralph** [comment]
Hi @ollie-acuto 👋 

Apologies for the late reply here. This must have slipped through the cracks.

Prisma does not currently support the exact functionality you're looking for. It does not support the `COALESCE` function or similar functionality to select fields from different models in a single query conditionally.

As you've mentioned, [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#:~:text=%24queryRaw%20does%20not%20support%20dynamic%20table%20names%20in%20PostgreSQL%20databases.%20Learn%20more) and `$queryRawUnsafe` could be used to execute raw SQL queries, but they do not support dynamic table names. 

The [client extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query) show that you can extend Prisma Client query operations, but it does not indicate that this can be done across multiple models or in conjunction with existing queries like findMany.

Can you please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase to help our engineers prioritize it?

**Comment by ollie-acuto** [comment]
Hi @ludralph, thanks for getting back to me and clarifying on this one!

OK, I'll go ahead and make a feature request
